op-node/cmd/genesis/cmd.go

```diff
@@ -175,13 +175,14 @@ var Subcommands = cli.Commands{
 	},
 }
 
-func makeRollupConfig(
-	config *genesis.DeployConfig,
-	l1StartBlock *types.Block,
-	l2Genesis *core.Genesis,
-	portalAddr common.Address,
-	sysConfigAddr common.Address,
-) *rollup.Config {
+func makeRollupConfig(config *genesis.DeployConfig, l1StartBlock *types.Block, l2Genesis *core.Genesis) (*rollup.Config, error) {
+	if config.OptimismPortalProxy == (common.Address{}) {
+		return nil, errors.New("OptimismPortalProxy cannot be address(0)")
+	}
+	if config.SystemConfigProxy == (common.Address{}) {
+		return nil, errors.New("SystemConfigProxy cannot be address(0)")
+	}
```

Comment:
shouldn't this be part of the deploy config `Check` function already?

Reply:
Yes it is, I could delete these checks but I wanted to be very sure that we don't build the rollup config incorrectly, ie a developer forgets to use a config file that includes these or they do not call the method to get the dev addresses or the addresses from disk using hardhat